src/pivot-table/components/cells/DimensionValue/Text.tsx

```diff
@@ -16,7 +16,11 @@ const Text = ({ children, cell, styleService, isCellSelected, isLeftColumn }: Pr
   <span
     style={{
       ...textStyle,
-      ...getLineClampStyle(isLeftColumn ? styleService.grid.lineClamp : DEFAULT_LINE_CLAMP),
+      ...getLineClampStyle(
+        isLeftColumn
+          ? Math.max(styleService.grid.lineClamp, cell.leafCount * styleService.grid.lineClamp)
```

Comment:
When leafCount is 0 fallback to the styleservice value

Reply:
Note: This logic is very much simplified. To ensure that the whole available cell height is used, the below logic would need to be used. Plus handle cell padding.

```js
const getLineClamp = ({ isLeftColumn, cell, styleService }: Pick<Props, "cell" | "isLeftColumn" | "styleService">) => {
  if (isLeftColumn) {
    if (cell.ref.qCanCollapse) {
      const textHeight = parseInt(styleService.dimensionValues.fontSize, 10) * LINE_HEIGHT_COEFFICIENT;
      const cellHeight = Math.max(styleService.contentCellHeight, cell.leafCount * styleService.contentCellHeight);

      return Math.floor((cellHeight / textHeight) * styleService.grid.lineClamp);
    }

    return styleService.grid.lineClamp;
  }

  return DEFAULT_LINE_CLAMP;
};
```